internal/buffer/search.go

```diff
@@ -21,33 +55,23 @@ func (b *Buffer) findDown(r *regexp.Regexp, start, end Loc) ([2]Loc, bool) {
 		start, end = end, start
 	}
 
-	for i := start.Y; i <= end.Y; i++ {
-		l := b.LineBytes(i)
-		charpos := 0
+	rPadNone, rPadStart, rPadEnd, rPadBoth := padRegexp(r)
+	rPadded := [4]*regexp.Regexp{rPadNone, rPadStart, rPadEnd, rPadBoth}
```

Comment:
I guess `padRegexp()` could just return the array of 4 regexps so we could just use it, and naturally index with the numeric constants `rPadNone`, `rPadStart`, `rPadEnd`, `rPadBoth` (defined with `iota` as I suggested above) instead of a separate variable for each.

And then, it also seems cleaner to let `findLineParams()` return not the regexp itself but just the numeric constant `rPadNone`, `rPadStart`, `rPadEnd` or `rPadBoth` (let's call it "padding mode"). Then we don't need to pass the array to `findLineParams()`.

Reply:
done